tests/ilc.cmd

```diff
@@ -15,41 +15,60 @@ goto :Loop
 
 :DoneArgs
 
-if not exist "%__SourceFolder%" goto :InvalidArgs
-if not exist "%__SourceFile%" goto :InvalidArgs
+if "%CoreRT_TestCompileMode%"=="" ((call :Fail "Test compile mode not set in CoreRT_TestCompileMode: Specify cpp/ryujit") & exit /b -1)
+set __RSPTemplate="%CoreRT_RspTemplateDir%\%CoreRT_TestCompileMode%.rsp"
+if not exist %__RSPTemplate% ((call :Fail "RSP template not found (%__RSPTemplate%)") & exit /b -1)
 
-REM ** Build variables
-if not defined CoreRT_BuildOS set CoreRT_BuildOS=Windows_NT
-if not defined CoreRT_BuildArch ((call :Fail "Set CoreRT_BuildArch to x86/x64/arm") & exit /b -1)
-if not defined CoreRT_BuildType ((call :Fail "Set CoreRT_BuildType to Debug or Release") & exit /b -1)
+set __RSPFile=%__SourceFile%.rsp
 
-if /i "%CoreRT_BuildType%"=="Debug" (
-    set __LinkLibs=msvcrtd.lib
-) else (
-    set __LinkLibs=msvcrt.lib
+rem Copy the relevant lines from the template RSP file
+findstr /V "\repro." %__RSPTemplate% > %__RSPFile%
+
+echo %__SourceFile% >> %__RSPFile%
+if exist "%__SourceFolder%\*.dll" (
+    echo -r:%__SourceFolder%\*.dll >> %__RSPFile%
 )
```

Comment:
I'd do `findstr "-r:" %__RSPTemplate% > %__RSPFile%` but I'll leave it to you if you are future proofing! Thanks for getting this off the floor.